analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CollectionQuerySimplification.cs

```diff
@@ -18,8 +18,7 @@ public CollectionQuerySimplification(List<object> coll)
             y = coll.Where(element => element is object).Select(element => (object)element); // Noncompliant use OfType
             x = coll.Where(element => element == null).Any();  // Noncompliant use Any([expression])
 //                   ^^^^^
-            var z = coll.Where(element => element == null).Count();  // Noncompliant {{Drop 'Where' and move the condition into the 'Count'.}}
-            z = Enumerable.Count(coll.Where(element => element == null));  // Noncompliant
+            var z = Enumerable.Count(coll.Where(element => element == null));  // Noncompliant
             z = Enumerable.Count(Enumerable.Where(coll, element => element == null));  // Noncompliant
```

Comment:
I left this UT test case with the intention to change the reason for which is non-compliant.
Now it's non-compliant because it should be written as `Count(predicate)`  (that based on https://github.com/SonarSource/sonar-dotnet/issues/4235 we should not propose since it's simple but less performant) and IMO it should change to being non-compliant because it's not really clear as it's written, it would be better to be `Enumerable.Where().Count()`. WDTYT?